client/my-sites/controller.js

```diff
@@ -183,6 +185,18 @@ module.exports = {
 				// if sites have loaded, but siteID is invalid, redirect to allSitesPath
 				if ( sites.select( siteID ) ) {
 					onSelectedSiteAvailable();
+				} else if ( ( currentUser.visible_site_count !== sites.getVisible().length ) ) {
+					ReactDom.render( (
+						<Main>
+							<JetpackManageErrorPage template="waitForJetpackSite" />
+						</Main>
+					), document.getElementById( 'primary' ) );
+
+					setTimeout( () => {
```

Comment:
We could do this without using timeouts, but relying 100% in events, if we extract the inline callback used on sites.once to its own function. Something line this, replacing lines 184-203:
```
const selectOnSitesChange = () => {
    // if sites have loaded, but siteID is invalid, redirect to allSitesPath                        
    if ( sites.select( siteID ) ) {
        onSelectedSiteAvailable();
    } else if ( ( currentUser.visible_site_count !== sites.getVisible().length ) ) {
        ReactDom.render( (
            <Main>
                <JetpackManageErrorPage template="waitForJetpackSite" />
            </Main>
        ), document.getElementById( 'primary' ) );

        sites.once( 'change', selectOnSitesChange );
        // We haven't set the main site, so we don't want to continue `next()`
        return;
    } else {
        page.redirect( allSitesPath );
    }
}

sites.once( 'change', selectOnSitesChange );
```